lib/dataStructures/queue.js

```diff
@@ -49,8 +49,8 @@
             return !this.isEmpty() ? this.elements[this.elements.length - 1] : null;
         };
 
-        self.getElements = function() {
-            return this.elements;
+        self.getElements = function() {

```

Comment:
can you please rename the function to `.toArray()` to avoid associating the name with the internal elements, also I believe it's cleaner to indicate that you want the queue converted to an array.

FYI: I'm almost finishing a new version of the repo that uses es6 for all the code, it'll be a major release, and will be done this week, and will include the additions that you are suggesting, stay tuned 👍